src/components/table/OcTableFiles.vue

```diff
@@ -388,6 +390,53 @@ export default {
 
       return this.$gettext("Select file")
     },
+
+    getSharedWithAvatarDescription(resource) {
+      const resourceType =
+        resource.type === "folder" ? this.$gettext("folder") : this.$gettext("file")
+
+      const shareCount = resource.sharedWith.filter(u => !u.link).length
+      const linkCount = resource.sharedWith.filter(u => !!u.link).length
+
+      const shareText =
+        shareCount > 0
+          ? this.$ngettext(
+              "This %{ resourceType } is shared via %{ shareCount } invite",
+              "This %{ resourceType } is shared via %{ shareCount } invites",
+              shareCount
+            )
+          : ""
+      const linkText =
+        linkCount > 0
+          ? this.$ngettext(
+              "This %{ resourceType } is shared via %{ linkCount } link",
+              "This %{ resourceType } is shared via %{ linkCount } links",
+              linkCount
+            )
+          : ""
+
+      const description = shareText + (shareCount > 0 && linkCount > 0 ? " " : "") + linkText
```

Comment:
Can be simplified to
```suggestion
      const description = [shareText, linkText].join(' ')
```

Reply:
One linter and unit test failure later this should actually be 🆗 🤒  (we're rendering a space now even if there's only one of the two present, but that's fine I suppose)

Reply:
With my solution (the `join`) the space doesn't happen if only one element is present.

Reply:
Tried it in the console (`test = ["one", ""].join(" ")` which returns `"one "`, also the snapshot test suggests it does so? Or am I missing sth?

Reply:
Ah. ups. Right 😅 You could add `trim()` to the result of `join(" ")`